protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -62,26 +124,139 @@ impl MovementClient {
 			.maptos_config
 			.client
 			.maptos_rest_connection_hostname;
-		let node_connection_port =
-			suzuka_config.execution_config.maptos_config.client.maptos_rest_connection_port;
+		let node_connection_port = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_rest_connection_port;
 
 		let node_connection_url =
 			format!("http://{}:{}", node_connection_address, node_connection_port);
 		let node_connection_url = Url::from_str(node_connection_url.as_str()).unwrap();
 
 		let rest_client = Client::new(node_connection_url.clone());
 
+		let faucet_listen_address = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_faucet_rest_connection_hostname
+			.clone();
+		let faucet_listen_port = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_faucet_rest_connection_port
+			.clone();
+
+		let faucet_connection_url = format!("http://{}:{}", node_connection_address, node_connection_port);
+		let faucet_listen_url = Url::from_str(faucet_connection_url.as_str()).unwrap();
+		let faucet_client = Arc::new(RwLock::new(FaucetClient::new(
+			faucet_listen_url.clone(),
+			node_connection_url.clone()
+		)));
+
 		let seed = [3u8; 32];
 		let mut rng = rand::rngs::StdRng::from_seed(seed);
 		let signer = LocalAccount::generate(&mut rng);
 
 		Ok(MovementClient {
+			counterparty_address: DUMMY_ADDRESS,
 			initiator_address: Vec::new(), //dummy for now
 			rest_client,
-			counterparty_address: DUMMY_ADDRESS,
+			faucet_client,
 			signer: Arc::new(signer),
 		})
 	}
+
+	pub async fn new_for_test(config: Config) -> Result<(Self, tokio::process::Child), anyhow::Error> {
+
+		let (setup_complete_tx, mut setup_complete_rx) = oneshot::channel();
+		let mut child = TokioCommand::new("aptos")
+		    .args(&["node", "run-local-testnet"])
+		    .stdout(Stdio::piped())
+		    .stderr(Stdio::piped())
+		    .spawn()?;
+	    
+		let stdout = child.stdout.take().expect("Failed to capture stdout");
+		let stderr = child.stderr.take().expect("Failed to capture stderr");
+	    
+		let node_handle = task::spawn(async move {
+		    let mut stdout_reader = BufReader::new(stdout).lines();
+		    let mut stderr_reader = BufReader::new(stderr).lines();
+
+
+			loop {
+				tokio::select! {
+					line = stdout_reader.next_line() => {
+						match line {
+							Ok(Some(line)) => {
+								println!("STDOUT: {}", line);
```

Comment:
We want to be using a tracing subscriber for tests. I've added a tracking issue, this can go into the feature branch as is https://github.com/movementlabsxyz/movement/issues/356